frontend/src/layout/Navbar.tsx

```diff
@@ -25,14 +26,21 @@ class _Navbar extends Component<Props> {
   getLoginState() {
     return (
       <div className='mr-2'>
-        {this.props.SessionData ? (
+        {this.props.SessionData.user ? (
           <div>
             <span className='text-lg font-bold mr-4'>{`Hello, ${this.props.SessionData.user.name}!`}</span>
             <button
               className='btn btn-sm btn-info mr-5'
               onClick={this.logoutSession}
             >
-              Logout
+              {this.props.loading ? (
+                <div className='flex items-center justify-center'>
+                  <div className='w-4 h-4 border-b-2 border-white-900 rounded-full animate-spin mr-5'></div>
+                  Logout
+                </div>
```

Comment:
Please use `<span/>` instead of `<div/>` according to w3c it is not valid.
check it in [w3c validator](https://validator.w3.org/nu/#textarea)